functions/src/modules/User.functions.ts

```diff
@@ -0,0 +1,214 @@
+import * as functions from 'firebase-functions';
+import * as admin from 'firebase-admin';
+import GoogleSheet from '../services/GoogleSheets';
+import { withDefault } from '../utils/parsers';
+import moment = require('moment');
+
+const db = admin.database();
+const userRoles = functions.database.ref('/users/{userId}/roles');
+
+enum RegistrationColumns {
+  Details = 'Details',
+  Status = 'Status',
+  Timestamp = 'Timestamp',
+  Name = 'Name',
+  Country = 'Country',
+  EmailAddress = 'Email Address',
+  PhoneNumber = 'Phone Number',
+  WhatsApp = 'WhatsApp',
+  Languages = 'Languages',
+  Services = 'Services',
+  Experience = 'Experience',
+  Influencer = 'Influencer',
+  RecommendedBy = 'Recommended By',
+}
+
+enum Roles {
+  CR = 'CR',
+  FC = 'FC',
+  QC = 'QC',
+  TE = 'TE',
+  SE = 'SE',
+  SQR = 'SQR',
+}
+
+enum Decision {
+  Yes = 'Yes',
+  No = 'No',
+}
+
+/**
+ * 1. Scan the Registrations Google Sheet and exports any new users to database
+ * 2. Do not import if user email already exists
+ *
+ */
+export const importUserRegistrationData = functions.https.onRequest(
+  async (req: functions.Request, res: functions.Response) => {
+    const gsheets: GoogleSheet = new GoogleSheet(
+      functions.config().registrations.spreadsheet_id,
+      'Registrations'
+    );
+
+    const registrationRows = await gsheets.getRows();
+
+    // Removes the header rows
+    registrationRows.splice(0, 1);
```

Comment:
Isn't it that `getRows()` return only data rows without headers? It should be the case. Our sheets service should take care of the headers.

@ahmad-elbatanouni please comment here as well.

Reply:
Yes, my bad, I mistakenly removed the line that removes the headers 
```javascript
this.headers = values.shift();
```
because it was removing the first row of each 1000 rows of each request.